controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -285,6 +286,31 @@ PravegaTablesScope newScope(final String scopeName) {
                 executor);
     }
 
+    @Override
+    public CompletableFuture<Void> addStreamTagsToIndex(String scope, String streamName, StreamConfiguration config,
+                                                        OperationContext ctx, Executor executor) {
+        OperationContext context = getOperationContext(ctx);
+        if (streamName.startsWith(NameUtils.INTERNAL_NAME_PREFIX)) {
+            // Tags are not allowed on internal streams.
+            return CompletableFuture.completedFuture(null);
```

Comment:
Should we throw an exception or at least log a warning that this was attempted?

Reply:
The methods `addStreamTagsToIndex` and `removeTagsFromIndex` are invoked as part of the Controller workflows during stream creation, deletion, or update for all streams.. This check ensures this step is skipped for internal streams.